rt_entt_codegen/core/print_sys_exec.cc

```diff
@@ -758,19 +760,48 @@ static auto print_ecsact_entt_system_details(
 
 		ctx.write("context.entity = entity;\n");
 
-		for(auto child_sys_id : get_child_system_ids(options.sys_like_id)) {
-			auto child_details = ecsact_entt_system_details::from_system_like(
-				ecsact_id_cast<ecsact_system_like_id>(child_sys_id)
-			);
+		auto child_system_ids = get_child_system_ids(options.sys_like_id);
 
-			auto child_system_name = cpp_identifier(decl_full_name(child_sys_id));
+		std::vector<ecsact_system_like_id> child_system_like_ids{};
 
-			ctx.write(
-				"ecsact::entt::execute_system<::" + child_system_name + ">(",
-				options.registry_var_name,
-				", &context);\n"
-			);
+		child_system_like_ids.resize(child_system_ids.size());
+
+		std::transform(
+			child_system_ids.begin(),
+			child_system_ids.end(),
+			child_system_like_ids.begin(),
+			[](auto system_id) {
+				return ecsact_id_cast<ecsact_system_like_id>(system_id);
+			}
+		);
+
+		if(!child_system_like_ids.empty()) {
+			if(child_system_ids.size() == 1) {
+				// TODO(Kelwan): Make use case system agnostic when we support
+				// nested Action systems
```

Comment:
created an issue for nested actions. Lets reference it here https://github.com/ecsact-dev/ecsact_parse/issues/154